code/game/objects/items/devices/drilldroptargetter.dm

```diff
@@ -0,0 +1,117 @@
+/obj/item/device/drill_dropper
+	name = "laser targeting dropper"
+	desc = "This drill literally pierces the heavens."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "drillpointer"
+	item_state = "binoculars"
+	slot_flags = SLOT_BELT
+	w_class = 2
+	var/has_dropped = 0
+	var/tileoffset = 8
+	var/viewsize = 7
+	var/emagged = FALSE
+
+/obj/item/device/drill_dropper/attack_self(mob/user)
+	zoom(user,tileoffset,viewsize)
+
+/obj/item/device/drill_dropper/attack(mob/living/M, mob/user)
+	laser_act(M, user)
+
+/obj/item/device/drill_dropper/afterattack(var/atom/target, var/mob/living/user, flag, params)
+	if(flag)	//we're placing the object on a table or in backpack
+		return
+	laser_act(target, user)
+
+/obj/item/device/drill_dropper/proc/laser_act(var/atom/target, var/mob/living/user)
+
+	if(has_dropped >= 2)
+		to_chat(user, "<span class='warning'>You can't use this device again!</span>")
+		return
+
+	var/turf/targloc = get_turf(target)
+	if(!emagged)
+		for(var/turf/t in block(locate(targloc.x+2,targloc.y+2,targloc.z),locate(targloc.x-2,targloc.y-2,targloc.z)))
+			if (!istype(t.loc, /area/mine))
+				to_chat(user, "<span class='warning'>You can't do this so close to the station, point the laser further into the mine!</span>")
+				return
+			if (!isfloor(targloc))
+				to_chat(user, "<span class='warning'>You cannot request a drill on unstable flooring!</span>")
+				return
+		for(var/turf/t2 in block(locate(targloc.x+2,targloc.y+2,targloc.z+1),locate(targloc.x-2,targloc.y-2,targloc.z+1)))
+			if (!istype(t2.loc, /area/mine))
+				to_chat(user, "<span class='warning'>You can't do this beneath the station, point the laser further away from it!</span>")
+				return
+	if(!(user in (viewers(14,target))) )
+		return
```

Comment:
Is it intended for the entire proc to not work if the user's not within 14 feet of the target? If so, it sounds like you might want a to_chat here too.

Reply:
Good call.